packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,201 @@
+<axis_data>(\r?\n)                        { this.popState(); return 'NEWLINE'; }
+<data>(\r?\n)                             { this.popState(); return 'NEWLINE'; }
+[\n\r]+                                   return 'NEWLINE';
+\%\%[^\n]*                                /* do nothing */
+
+title                                     { this.begin("title"); return 'title'; }
+<title>(?!\n|;|#)*[^\n]*                  { this.popState(); return "title_value"; }
+
+
+accTitle\s*":"\s*                         { this.begin("acc_title");return 'acc_title'; }
+<acc_title>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_title_value"; }
+accDescr\s*":"\s*                         { this.begin("acc_descr");return 'acc_descr'; }
+<acc_descr>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_descr_value"; }
+accDescr\s*"{"\s*                         { this.begin("acc_descr_multiline");}
+<acc_descr_multiline>[\}]                 { this.popState(); }
+<acc_descr_multiline>[^\}]*               { return "acc_descr_multiline_value"; }
+
+"xychart-beta"		                        {return 'XYCHART';}
+("vertical"|"horizontal")                 {return 'CHART_ORIENTATION'}
+
+"x-axis"                                  { this.begin("axis_data"); return "X_AXIS"; }
+"y-axis"                                  { this.begin("axis_data"); return "Y_AXIS"; }
+<axis_data>[\[]                           { this.popState(); return 'SQUARE_BRACES_START'; }
+<axis_data>[+-]?\d+(?:\.\d+)?             { return 'NUMBER_WITH_DECIMAL'; }
+<axis_data>"-->"                          { return 'ARROW_DELIMITER'; }
+
+
+"line"                                    { this.begin("data"); return 'LINE'; }
+"bar"                                     { this.begin("data"); return 'BAR'; }
+<data>[\[]                                { this.popState(); this.begin("data_inner"); return 'SQUARE_BRACES_START'; }
+<data_inner>[+-]?\d+(?:\.\d+)?            { return 'NUMBER_WITH_DECIMAL';}
+<data_inner>[\]]                          { this.popState(); return 'SQUARE_BRACES_END'; }
+
+
+
+
+["][`]                                    { this.begin("md_string");}
+<md_string>[^`"]+                         { return "MD_STR";}
+<md_string>[`]["]                         { this.popState();}
+["]                                       this.begin("string");
+<string>["]                               this.popState();
+<string>[^"]*                             return "STR";
+
+
+[\[]                                      return 'SQUARE_BRACES_START'
+[\]]                                      return 'SQUARE_BRACES_END'
+[A-Za-z]+                                 return 'ALPHA';
+":"                                       return 'COLON';
+\+                                        return 'PLUS';
+","                                       return 'COMMA';
+"="                                       return 'EQUALS';
+\=                                        return 'EQUALS';
```

Comment:
It seems that these 2 rules are equal, so one of them is redundant

Reply:
Thanks! removed